state/azure/tablestorage/tablestorage.go

```diff
@@ -241,63 +263,94 @@ func (r *StateStore) writeRow(req *state.SetRequest) error {
 }
 
 func isNotFoundError(err error) bool {
-	azureError, ok := err.(storage.AzureStorageServiceError)
-
-	return ok && azureError.Code == "ResourceNotFound"
+	var respErr *azcore.ResponseError
+	if errors.As(err, &respErr) {
+		return (respErr.ErrorCode == string(aztables.ResourceNotFound)) || (respErr.ErrorCode == string(aztables.EntityNotFound))
+	}
+	return false
 }
 
 func isEntityAlreadyExistsError(err error) bool {
-	azureError, ok := err.(storage.AzureStorageServiceError)
-
-	return ok && azureError.Code == "EntityAlreadyExists"
+	var respErr *azcore.ResponseError
+	if errors.As(err, &respErr) {
+		return respErr.ErrorCode == string(aztables.EntityAlreadyExists)
+	}
+	return false
 }
 
 func isTableAlreadyExistsError(err error) bool {
-	azureError, ok := err.(storage.AzureStorageServiceError)
-
-	return ok && azureError.Code == "TableAlreadyExists"
+	var respErr *azcore.ResponseError
+	if errors.As(err, &respErr) {
+		return respErr.ErrorCode == string(aztables.TableAlreadyExists)
+	}
+	return false
 }
 
 func (r *StateStore) deleteRow(req *state.DeleteRequest) error {
-	pk, rk := getPartitionAndRowKey(req.Key)
-	entity := r.table.GetEntityReference(pk, rk)
+	pk, rk := getPartitionAndRowKey(req.Key, r.cosmosDBmode)
 
 	if req.ETag != nil {
-		entity.OdataEtag = *req.ETag
-
-		// force=false sets the "If-Match: <ETag>" header to ensure that the delete is only performed if the
-		// entity's ETag matches the specified ETag
-		return entity.Delete(false, nil)
+		azcoreETag := azcore.ETag(*req.ETag)
+		_, err := r.client.DeleteEntity(ctx.Background(), pk, rk, &aztables.DeleteEntityOptions{IfMatch: &azcoreETag})
+		return err
 	}
-
-	// force=true sets the "If-Match: *" header to ensure that we delete a matching entity
-	// regardless of the entity's ETag value
-	return entity.Delete(true, nil)
+	all := azcore.ETagAny
+	_, err := r.client.DeleteEntity(ctx.Background(), pk, rk, &aztables.DeleteEntityOptions{IfMatch: &all})
+	return err
 }
 
-func getPartitionAndRowKey(key string) (string, string) {
+func getPartitionAndRowKey(key string, cosmosDBmode bool) (string, string) {
 	pr := strings.Split(key, keyDelimiter)
 	if len(pr) != 2 {
-		return pr[0], ""
+		if cosmosDBmode {
+			return pr[0], "_dapr_empty_row_key_value_"
+		} else {
+			return pr[0], ""
+		}
 	}
 
 	return pr[0], pr[1]
 }
 
-func (r *StateStore) marshal(req *state.SetRequest) string {
-	var v string
+func (r *StateStore) marshal(req *state.SetRequest) ([]byte, error) {
+	var value string
 	b, ok := req.Value.([]byte)
 	if ok {
-		v = string(b)
+		value = string(b)
 	} else {
-		v, _ = jsoniter.MarshalToString(req.Value)
+		value, _ = jsoniter.MarshalToString(req.Value)
 	}
 
-	return v
+	pk, rk := getPartitionAndRowKey(req.Key, r.cosmosDBmode)
+
+	entity := aztables.EDMEntity{
+		Entity: aztables.Entity{
+			PartitionKey: pk,
+			RowKey:       rk,
+		},
+		Properties: map[string]interface{}{
+			valueEntityProperty: value,
+		},
+	}
+
+	var etag string
```

Comment:
Nit. Can probably just be this?

```go
if req.ETag != nil {
	entity.ETag = *req.ETag
}
```

Reply:
Yeah - this was left over from some refactor. Updated now.